MetaMask/metamask-extension
# Allow background to handle forgotten password flow

## Explanation
Prior to this change the way the forgot password flow from the lock screen works is:
1. Call the markPasswordForgotten thunk in actions.js.
2. In the background, we'd set the forgottenPassword key on preferences controller.
3. The forgottenPassword key gets flattened into the metamask state object.
4. The new state change gets sent to the UI and redux
5. Simultaneously we'd dispatch the FORGOT_PASSWORD action.
6. This would set the forgottenPassword key in appState to true
7. In the home.container.js file we'd get the forgottenPassword value from app, but in the routes.container we get it from the background ('metamask') state. 
8. The home page would route to the forgot password screen. 
9. If the account is reset OR the user successfully logs in we'd call the unMarkPasswordForgotten method which would call into the background to set the preference then an action was dispatched to change the app state to match.

This change eliminates the duplication of state in the app slice and makes a simple change to mark the forgottenPassword preference controller state as false if the wallet is successfully unlocked (from the background) this change allows us to remove the FORGOT_PASSWORD action entirely. 

## Manual Testing Steps
1. pull this branch.
2. run yarn
3. `yarn start`
4. At lock screen click forgot password
5. See the restore wallet screen and enter seed phrase
6. See your wallet, no further routing to the restore screen.
7. Repeat process but instead of restoring go back to the lock screen and type password 
8. Once unlocked see no further routing to restore screen.
9. You may also verify that the background state changes by using redux devtools to inspect actions.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
